packages/components/src/menu-item/test/index.js

```diff
@@ -72,4 +72,26 @@ describe( 'MenuItem', () => {
 
 		expect( wrapper.prop( 'aria-label' ) ).toBeUndefined();
 	} );
+
+	it( 'should avoid using aria-checked if only menuitem is set as aria-role', () => {
+		const wrapper = shallow(
+			<MenuItem role="menuitem"><div /></MenuItem>
+		);
+
+		expect( wrapper.prop( 'aria-checked' ) ).toBeUndefined();
```

Comment:
I don't think this test is quite right since it would have passed with the previous code as well (aria-checked would be undefined whenever isSelected is undefined), so there's no assertion that the behaviour has changed. The MenuItem should have an isSelected prop to demonstrate that the prop is not included in the rendered output.